components/segmenter/src/complex/mod.rs

```diff
@@ -17,33 +16,42 @@ mod lstm;
 #[cfg(feature = "lstm")]
 use lstm::*;
 
-#[cfg(not(feature = "lstm"))]
-type DictOrLstm = Either<DataPayload<UCharDictionaryBreakDataV1>, core::convert::Infallible>;
-#[cfg(not(feature = "lstm"))]
-type DictOrLstmBorrowed<'a> = Either<&'a UCharDictionaryBreakData<'a>, core::convert::Infallible>;
+#[derive(Debug, Clone)]
+#[allow(clippy::large_enum_variant)]
+enum DictOrLstm {
+    Dict(DataPayload<UCharDictionaryBreakDataV1>),
+    #[cfg(not(feature = "lstm"))]
+    Lstm(core::convert::Infallible),
```

Comment:
Suggestion: delete the variant if it is uninhabited.

Reply:
Will try to fix in my followup

Reply:
Fix in https://github.com/unicode-org/icu4x/pull/6409